---
title: "Bioinformatics spring school"
collection: teaching
type: "Week-long workshop"
permalink: /teaching/Bioinformatics_Spring_School

date: 2020-12
location: "Dunedin, New Zealand"
---

Bioinformatics spring school website]
(https://otagocarpentries.github.io/bioinformatics-spring-school-2020/#curriculum)


### Bioinformatic topics


#### DNA variant calling from next generation sequence data

- Project organisation
- Assessing read quality
- Trimming and filtering
- Variant calling workflow
- Automating a variant calling workflow


#### Genotype-by-sequencing analysis (RAD-Sequencing)
 
- Data quality assessment
- Calling variant workflow using Stacks (with or without a reference genome)
- Basic population genetic analyses (population structure, phylogenetic tree) 

#### RNA-sequencing differential expression analysis

- Quality assessment
- Trimming and filtering
- Mapping and count
- Differential expression
- Overrepresentation analysis

#### Environmental DNA analysis
 
- Next-generation sequencing approaches for environmental samples (metabarcoding)
- Demultiplexing and quality assessment of NGS amplicon data
- Clustering and denoising of sequence data
- Determine species composition of samples (taxonomic assignment)
- community ecology statistics for eDNA results

### Computational Topics

These topics will be integrated into the bioinformatics workshops and built upon across the week with the goal of demostrating a best practices workflow approach to Bioinformatics analysis.

#### Introduction to the Unix command line (Bash)

Many bioinformatic programs will only operate in a Unix command line environment, as such we need to provide an introduction to working in this environment which will cover:

- Navigating around the filesystem (files/directories/folders)
- Running command line based programs
- Creating scripts to automate workflows

#### Introduction to R

Analysis portions of the workshops will be performed using R. As such we will provide a brief introduction to R. This will include:

- How to interact with R
- Creating objects
- How to use functions
- How to install and use packages

#### Introduction to working in a high performance computing environment

We'll be making use of the New Zealand eScience Infrastructure (NeSI), which is the national provider of high performance computing for researchers, to run our analysis and as part of this we'll cover:

- What/who is NeSI
- Logging onto a remote computer
- Transfer of data to/from a remote computer
- What is a job scheduler and how to use it
- How to request a project allocation on NeSI


#### Introduction to version control using git

Running programs on the command line gives us the opportunity to save the commands used in scripts that we can reuse later on. It's important that as we develop these scripts that we have a machanism to track the changes made. Git is a program that enables us to track the changes we make and see the differences between previous versions. The introduction to version control will cover:

- Setting up Git
- Using git to track changes on scripts
- How to view changes between versions of scripts
- Use git to look at previous versions of scripts
