---
title: "vcf_tools.py"
collection: code
permalink: /code/vcf_tools
citation: '<b>vcf_tools.py/</b>.
Python library used to process VCFs during my PhD. Some of the main functions:
* extract_pi_double_vcf_bed() Append pi information line by line to a bed file. It uses two vcfs as input ( one for genotypes, and one for depth at all sites)
* count_sites_under_condition_vcf() Check the number of sites that respect a condition for a sub part of a vcf file.
* all_freq_spectrum_vcf_bed() Append the folded allele frequency spectrum information line by line to a bed file filtering the vcf file according to user based conditions. It uses two vcfs as input ( one for genotypes, and one for depth at all sites)( one for genotypes, and one for depth at all sites)'
githuburl: 'https://github.com/ldutoit/personal_libraries/blob/master/vcf_tools.py'
---




